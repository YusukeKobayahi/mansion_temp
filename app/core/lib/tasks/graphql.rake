# frozen_string_literal: true

module GraphqlTask
  extend Rake::DSL

  namespace :graphql do
    namespace :consumer do
      task schema_dump: :environment do
        require 'graphql/rake_task'

        GraphQL::RakeTask.new(
          load_schema: lambda { |_task|
            require File.expand_path('../../app/graphql/consumer/inno_schema', __dir__)
            Consumer::InnoSchema
          },
          directory: './graphql/consumer',
        )
        Rake::Task['graphql:schema:idl'].invoke
      end
    end
  end
end
