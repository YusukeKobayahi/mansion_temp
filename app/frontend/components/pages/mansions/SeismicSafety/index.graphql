fragment MsLibraryMansionsSeismicSafety on Mansion {
  name
  age
  constructedIn
  structure {
    name
    slug
  }
}
