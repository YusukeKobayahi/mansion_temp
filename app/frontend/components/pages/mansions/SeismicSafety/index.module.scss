@import "~/styles/utils";

@import "~/styles/utils";

.container {
  margin-top: 50px;
  padding: 0 40px;

  @include screenMd {
    padding: 0 16px;
  }
}

.info {
  line-height: 1.5;
}

.table {
  width: 100%;
  border-bottom: 1px solid #d0d0d0;
  margin-top: 32px;

  td {
    padding: 15px;
    border-top: 1px solid #d0d0d0;
    line-height: 1.5;
    background-clip: padding-box;
    vertical-align: middle;
  }

  th {
    width: 144px;
    padding: 15px;
    border-top: 1px solid #d0d0d0;
    background-color: #faf9f7;
    line-height: 1.5;
    text-align: center;
    vertical-align: middle;

    // firefoxで枠線が表示されない問題の対応
    // See: https://qiita.com/h-koeda/items/3b4c6a465ff4f81c16dc
    background-clip: padding-box;

    img {
      display: block;
      margin: 0 auto 6px;
    }

    &.hasSafety {
      position: relative;
      background-color: #edf1f4;

      &::before {
        content: "この物件";
        display: block;
        position: absolute;
        left: 6px;
        top: 50%;
        transform: translateY(-50%); // 縦方向に中央表示
        width: 22px;
        padding: 3px 0;
        border: 1px solid #707070;
        background-color: #ffffff;
        font-size: 12px;
        line-height: 1;
        text-align: center;
      }
    }
  }

  @include screenMd {
    td {
      display: block;
      width: 100%;
      padding: 11px 9px;
    }

    th {
      display: block;
      width: 100%;
      padding: 6px;

      &.hasSafety {
        &::before {
          left: auto;
          right: 7px;
          width: auto;
          padding: 3px;
        }
      }
    }
  }
}
