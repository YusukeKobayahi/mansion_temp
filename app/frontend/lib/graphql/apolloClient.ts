// 参考
// https://github.com/vercel/next.js/tree/canary/examples/with-apollo

import { useMemo } from "react";
import { ApolloClient, ApolloLink, InMemoryCache } from "@apollo/client";
import { createHttpLink } from "apollo-link-http";
import { from } from "apollo-link";
import { createFragmentArgumentLink } from "apollo-link-fragment-argument";
import { mergeDeep, relayStylePagination } from "@apollo/client/utilities";
import { isSSR } from "~/lib/utils";

/* eslint-disable @typescript-eslint/no-explicit-any -- キャッシュの型を事前に決められないため */
type CacheState = Record<string, any> | null;
/* eslint-enable @typescript-eslint/no-explicit-any */

/* eslint-disable @typescript-eslint/naming-convention -- GraphQLのタイプ名 (このconventionに従わない)を指定したいので */
const createCache = () =>
  new InMemoryCache({
    typePolicies: {
      Mansion: {
        keyFields: ["uniqueCode"],
        fields: {
          salesHistories: relayStylePagination(),
        },
      },
    },
  });
/* eslint-enable @typescript-eslint/naming-convention */

const link = from([
  createFragmentArgumentLink(),
  createHttpLink({
    uri: `${process.env.NEXT_PUBLIC_CORE_API_URL}/consumer/graphql`,
    fetchOptions: {
      mode: "cors",
    },
  }),
]);

function createApolloClient() {
  return new ApolloClient({
    ssrMode: isSSR(),
    // See: https://github.com/apollographql/apollo-link/issues/1258#issuecomment-604031488
    // TODO: @apollo/clientバージョン上げたら直るかも
    link: (link as unknown) as ApolloLink,
    cache: createCache(),
    defaultOptions: {
      watchQuery: {
        // SSR時にuseQueryによってリクエストしてレンダリングする場合、
        // getServerSidePropsからキャッシュが渡っている必要がある。(でないと描画されない)
        // そのため、SSR時は常にcacheから取得されるべきで、実装意図を明確にするためPolicyも設定しておく
        // (エラーになるわけではない)
        fetchPolicy: isSSR() ? "cache-only" : "cache-first",
      },
    },
  });
}

let apolloClient: ReturnType<typeof createApolloClient>;

export function initializeApollo(
  initialState: CacheState = null
): ApolloClient<CacheState> {
  const _apolloClient = apolloClient ?? createApolloClient();

  // If your page has Next.js data fetching methods that use Apollo Client, the initial state
  // gets hydrated here
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();

    // Merge the existing cache into data passed from getStaticProps/getServerSideProps
    const data = mergeDeep(initialState, existingCache);

    // Restore the cache with the merged data
    _apolloClient.cache.restore(data);
  }
  // For SSG and SSR always create a new Apollo Client
  if (isSSR()) return _apolloClient;
  // Create the Apollo Client once in the client
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(initialState: CacheState): ApolloClient<CacheState> {
  const store = useMemo(() => initializeApollo(initialState), [initialState]);
  return store;
}
