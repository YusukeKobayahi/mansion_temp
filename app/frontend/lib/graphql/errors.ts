export interface GraphQLError {
  graphQLErrors?: {
    message: string;
    extensions?: {
      code: string;
    };
  }[];
}

export const isNotFoundError = (err: GraphQLError): boolean => {
  const graphQLErrors = err.graphQLErrors;
  if (graphQLErrors == null) return false;

  return graphQLErrors[0]?.extensions?.code === "NOT_FOUND";
};
