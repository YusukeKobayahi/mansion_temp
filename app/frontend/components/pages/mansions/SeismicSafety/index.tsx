import Image from "next/image";
import { MsLibraryMansionsSeismicSafetyFragment } from "~/graphql/generated";
import SidelineHeading from "~/components/commons/headings/SidelineHeading";
import styles from "./index.module.scss";

export interface SeismicSafetyProps {
  data: MsLibraryMansionsSeismicSafetyFragment;
}

const SeismicSafety: React.FC<SeismicSafetyProps> = ({
  data,
}: SeismicSafetyProps) => {
  const hasSafty = Date.parse(data.constructedIn) < Date.parse("1983-06-01");

  return (
    <section className={styles.container}>
      <h3>
        <SidelineHeading>耐震性</SidelineHeading>
      </h3>
      <p className={styles.info}></p>
      <table className={styles.table}>
        <tbody>
          <tr>
            <th className={hasSafty ? styles.hasSafety : ""}>
              <div className={styles.image}>
                <Image
                  src={"/ms-library/images/old-standard.svg"}
                  alt={"旧耐性基準"}
                  width={35}
                  height={36}
                />
              </div>
              旧耐性基準
            </th>
            <td>
              中規模の地震（震度５強）程度が発生した際に「建物が倒壊しないこと」を基準に定められています。
              <br />
              耐震性が気になるのであれば、耐震補強工事などを検討してみても良いでしょう。
            </td>
          </tr>
          <tr>
            <th className={!hasSafty ? styles.hasSafety : ""}>
              <div className={styles.image}>
                <Image
                  src={"/ms-library/images/new-standard.svg"}
                  alt={"新耐性基準"}
                  width={40}
                  height={36}
                />
              </div>
              新耐性基準
            </th>
            <td>
              新耐震基準では中規模の地震（震度5強程度）でほとんど損傷せず、大規模な地震震度6強～7程度の揺れでも倒壊しないような構造基準が適用されており、安心して住める基準を満たしています。
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  );
};

export default SeismicSafety;
