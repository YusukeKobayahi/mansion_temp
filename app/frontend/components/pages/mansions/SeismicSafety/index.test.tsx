import { render } from "@testing-library/react";
import mockData from "../fixtures/mockData.json";
import { MansionQuery } from "~/graphql/generated";
import SeismicSafety from ".";

describe("components/pages/mansions/SeismicSafety", () => {
  it("正常に表示される", () => {
    const mock = mockData as MansionQuery;
    const { asFragment } = render(<SeismicSafety data={mock.mansion} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("築年月が1983年5月以前だったら何も表示されない", () => {
    const data = {
      name: "ガーラ・レジデンス三鷹シャインパークス",
      age: 37,
      constructedIn: "1983-05-31",
      structure: {
        name: "鉄骨造(S造)",
        slug: "steel",
      },
    };
    const { asFragment } = render(<SeismicSafety data={data} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("築年月が1983年6月以降だったら表示される", () => {
    const data = {
      name: "ガーラ・レジデンス三鷹シャインパークス",
      age: 37,
      constructedIn: "1983-06-01",
      structure: {
        name: "鉄骨造(S造)",
        slug: "steel",
      },
    };
    const { asFragment } = render(<SeismicSafety data={data} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
