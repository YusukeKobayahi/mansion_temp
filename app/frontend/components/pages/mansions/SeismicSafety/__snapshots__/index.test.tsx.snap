// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/mansions/SeismicSafety 正常に表示される 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <h3>
      <div
        class="heading"
      >
        <span
          class="inner"
        >
          耐震性
        </span>
      </div>
    </h3>
    <p
      class="info"
    />
    <table
      class="table"
    >
      <tbody>
        <tr>
          <th
            class=""
          >
            <div
              class="image"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzUiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <noscript />
                <img
                  alt="旧耐性基準"
                  data-nimg="true"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            旧耐性基準
          </th>
          <td>
            中規模の地震（震度５強）程度が発生した際に「建物が倒壊しないこと」を基準に定められています。
            <br />
            耐震性が気になるのであれば、耐震補強工事などを検討してみても良いでしょう。
          </td>
        </tr>
        <tr>
          <th
            class="hasSafety"
          >
            <div
              class="image"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <noscript />
                <img
                  alt="新耐性基準"
                  data-nimg="true"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            新耐性基準
          </th>
          <td>
            新耐震基準では中規模の地震（震度5強程度）でほとんど損傷せず、大規模な地震震度6強～7程度の揺れでも倒壊しないような構造基準が適用されており、安心して住める基準を満たしています。
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</DocumentFragment>
`;

exports[`components/pages/mansions/SeismicSafety 築年月が1983年5月以前だったら何も表示されない 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <h3>
      <div
        class="heading"
      >
        <span
          class="inner"
        >
          耐震性
        </span>
      </div>
    </h3>
    <p
      class="info"
    />
    <table
      class="table"
    >
      <tbody>
        <tr>
          <th
            class="hasSafety"
          >
            <div
              class="image"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzUiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <noscript />
                <img
                  alt="旧耐性基準"
                  data-nimg="true"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            旧耐性基準
          </th>
          <td>
            中規模の地震（震度５強）程度が発生した際に「建物が倒壊しないこと」を基準に定められています。
            <br />
            耐震性が気になるのであれば、耐震補強工事などを検討してみても良いでしょう。
          </td>
        </tr>
        <tr>
          <th
            class=""
          >
            <div
              class="image"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <noscript />
                <img
                  alt="新耐性基準"
                  data-nimg="true"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            新耐性基準
          </th>
          <td>
            新耐震基準では中規模の地震（震度5強程度）でほとんど損傷せず、大規模な地震震度6強～7程度の揺れでも倒壊しないような構造基準が適用されており、安心して住める基準を満たしています。
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</DocumentFragment>
`;

exports[`components/pages/mansions/SeismicSafety 築年月が1983年6月以降だったら表示される 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <h3>
      <div
        class="heading"
      >
        <span
          class="inner"
        >
          耐震性
        </span>
      </div>
    </h3>
    <p
      class="info"
    />
    <table
      class="table"
    >
      <tbody>
        <tr>
          <th
            class=""
          >
            <div
              class="image"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzUiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <noscript />
                <img
                  alt="旧耐性基準"
                  data-nimg="true"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            旧耐性基準
          </th>
          <td>
            中規模の地震（震度５強）程度が発生した際に「建物が倒壊しないこと」を基準に定められています。
            <br />
            耐震性が気になるのであれば、耐震補強工事などを検討してみても良いでしょう。
          </td>
        </tr>
        <tr>
          <th
            class="hasSafety"
          >
            <div
              class="image"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <noscript />
                <img
                  alt="新耐性基準"
                  data-nimg="true"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            新耐性基準
          </th>
          <td>
            新耐震基準では中規模の地震（震度5強程度）でほとんど損傷せず、大規模な地震震度6強～7程度の揺れでも倒壊しないような構造基準が適用されており、安心して住める基準を満たしています。
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</DocumentFragment>
`;
